src/main/java/carpet/helpers/HopperCounter.java

```diff
@@ -1,5 +1,6 @@
 package carpet.helpers;
 
+import carpet.utils.WoolTool;
```

Comment:
This part is completely broken, throws NPE's and fails to display stuff. Did you run it?

Reply:
k, fixed now